### PR TITLE
feat(patterns): add Error documentation

### PR DESCRIPTION
**Pull request by steue**
## Description

Documentation for Error handling

## Detail

This page goes into detail on how to design & format error messages through our Global Alert, Alert, Notifications, and Validation message components. For increased readability, this page slightly deviates from other pattern documentation structure by divvying up the components and detailing the patterns of each

p.s. there might be some additional image/formatting/content changes, just want to get this PR up

p.p.s. this is my first website PR plz be nice, but most importantly plz help me make sure i dont break anything ˙ ͜ʟ˙ 

## Checklist

- [ ] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [ ] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [ ] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/patterns/errors.mdx [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+Error messages are best surfaced using these components:
+
+- Global Alerts
+- Alerts
+- Notifications
+- Validation messages (in Forms)
+
+## Global Alerts
+
+Use the Global Alert to communicate system level errors that interrupt a user’s task.
```
Comment (Francois-Esquire) [inline src/pages/patterns/errors.mdx:21]:
```suggestion
Use the Global Alert to communicate system level errors that interrupt a users task.
```
I don't think contraction was intended here.

File: src/pages/patterns/errors.mdx [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+- The system has failed to load or perform an action
+- The user has made an incorrect action or input
+
+Error messages are best surfaced using these components:
+
+- Global Alerts
+- Alerts
+- Notifications
+- Validation messages (in Forms)
```
Comment (Francois-Esquire) [inline src/pages/patterns/errors.mdx:17]: Would it be helpful to link to these components?

**Review by Francois-Esquire — CHANGES_REQUESTED** [review]

File: src/pages/patterns/errors.mdx (hunk 1 above)
Comment (jzempel) [reply src/pages/patterns/errors.mdx:21]: I think the apostrophe needs to remain here. https://forum.wordreference.com/threads/user%E2%80%99s-users-or-users.1474044/#post-7448965

Comment (Francois-Esquire) [reply src/pages/patterns/errors.mdx:21]: Good call @jzempel

File: src/pages/patterns/errors.mdx [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    </ul>
+  </Dont>
+</BestPractice>
+
+## Accessibility
+
+Error messages need to be perceivable and understandable for people of all abilities.
+An accessible error message should exhibit these qualities.
+
+- Written in text. Icons and color can be used, but not alone
```
Comment (jzempel) [inline src/pages/patterns/errors.mdx:181]:
```suggestion
- Written in text. Icons and color can be used, but not alone.
```

[nit] per https://garden.zendesk.com/content/punctuation#period

File: src/pages/patterns/errors.mdx [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+### Global alert anatomy
+
+- Always include a title to explain what has went wrong and how to fix it
+- Provide a clear action to address the issue
+
+![Global Alert anatomy](figma:patterns-error-global-alert-anatomy.png)
+
+1. Title
+2. Message
+3. Button
```
Comment (jzempel) [inline src/pages/patterns/errors.mdx:32]:
```suggestion
1. Title
1. Message
1. Button
```

[nit] the suggested markdown follows Garden's preference for ordered lists. Consider applying throughout.

File: src/pages/patterns/errors.mdx [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+## Additional resources
+
+[https://www.w3.org/WAI/standards-guidelines/act/rules/36b590/proposed/](https://www.w3.org/WAI/standards-guidelines/act/rules/36b590/proposed/)
+[https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html)
+[https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html](https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html)
+
+---
+
+<h2>Feedback</h2>
```
Comment (jzempel) [inline src/pages/patterns/errors.mdx:194]:
```suggestion
## Feedback
```

**Review by jzempel — CHANGES_REQUESTED** [review]

File: src/pages/patterns/errors.mdx [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+- The system has failed to load or perform an action
+- The user has made an incorrect action or input
+
+Error messages are best surfaced using these components:
+
+- [Global Alerts](/components/global-alert)
+- [Alerts](/components/alerts)
+- [Notifications](/components/notifications)
+- [Validation messages](/components/forms/input#validation)
```
Comment (jzempel) [inline src/pages/patterns/errors.mdx:19]:
```suggestion
- [Validation messages](/components/input#validation)
```

**Review by jzempel — CHANGES_REQUESTED** [review]

**Review by Francois-Esquire — APPROVED** [review]

**Review by jzempel — APPROVED** [review]
nice, @steue 🎉

**Pull request merged**